src/core/dir.c

```diff
@@ -46,14 +46,16 @@ static const char *filename_to_utf8(const wchar_t *str)
 #define CURRENT_DIR "."
 #endif
 
+#define MAX_FILE_NAME_BUFFER (FILE_NAME_MAX * 2)
+
 static dir_listing listing;
 static int listing_initialized = 0;
 
 static void clear_dir_listing(void)
 {
     if (!listing_initialized) {
         for (int i = 0; i < DIR_MAX_FILES; i++) {
-            listing.files[i] = malloc(FILE_NAME_MAX * sizeof(char));
+            listing.files[i] = malloc(MAX_FILE_NAME_BUFFER * sizeof(char));
```

Comment:
What was the warning here? Allocating twice the amount of memory seems overkill.

Reply:
This one was problematic.

The Switch and Vita builds were reporting this warning:

```
/build/git/src/core/dir.c:83:13: warning: 'strncpy' output may be truncated copying 199 bytes from a string of length 255 [-Wstringop-truncation]
             strncpy(listing.files[listing.num_files], name, FILE_NAME_MAX - 1);
```

The problem is that `dirent.h` does not define a standard size for its member `d_name` in the `dirent` structure. In some platforms its size is set to `200`, and in others to `255` or even `260`.

Since `FILE_NAME_MAX` is always `200`, hence the problem with that warning. Also, there is no platform-specific macro for the `d_name` string length. Therefore I decided to create a new macro, making it twice as large as `FILE_NAME_MAX` for convenience.

However it could probably be set to `300` and still work fine.

Reply:
Would increasing `FILE_NAME_MAX` to 300 help? I was looking at the UTF-8 encoding/decoding functions today, and having a filename full of "weird" characters would go over the limit.

Reply:
> Would increasing `FILE_NAME_MAX` to 300 help? 

Yes it would!

> I was looking at the UTF-8 encoding/decoding functions today, and having a filename full of "weird" characters would go over the limit.

Yeah, UTF-8 can take up to 4 bytes per char, especially on special glyphs like chinese characters... I wonder what the char limit for chinese windows is!

Reply:
> > Would increasing FILE_NAME_MAX to 300 help?

> Yes it would!

Then please do :smile:

> I wonder what the char limit for chinese windows is!

Still 255, since Windows uses 16-bit unicode for filenames. We only get them as UTF-8 because of the compatibility `dirent.h` file, which is there because Windows does not support opendir natively.

Reply:
Done!

I didn't think `FILE_NAME_MAX` was set by Julius, but then again I didn't really check 😆